python_modules/dagster/dagster/_core/definitions/repository_definition/repository_data_builder.py

```diff
@@ -263,30 +260,22 @@ def build_caching_repository_data_from_list(
             name,
             unresolved_partitioned_asset_schedule,
         ) in unresolved_partitioned_asset_schedules.items():
-            resolved_job = pipelines_or_jobs[unresolved_partitioned_asset_schedule.job.name]
+            resolved_job = jobs[unresolved_partitioned_asset_schedule.job.name]
             schedules[name] = unresolved_partitioned_asset_schedule.resolve(
                 cast(JobDefinition, resolved_job)
             )
 
-    for pipeline_or_job in pipelines_or_jobs.values():
-        pipeline_or_job.validate_resource_requirements_satisfied()
-
-    pipelines: Dict[str, PipelineDefinition] = {}
-    jobs: Dict[str, JobDefinition] = {}
-    for name, pipeline_or_job in pipelines_or_jobs.items():
-        if isinstance(pipeline_or_job, JobDefinition):
-            jobs[name] = pipeline_or_job
-        else:
-            pipelines[name] = pipeline_or_job
+    for job_def in jobs.values():
+        job_def.validate_resource_requirements_satisfied()
 
     if default_executor_def:
         for name, job_def in jobs.items():
-            if not job_def._executor_def_specified:  # noqa: SLF001
+            if not job_def.has_specified_executor:
```

Comment:
one subtle behavior change here is that after defaults from the definitions object / repo have been applied, `has_specified_executor` will be true on the job definition within the repo, whereas I think it wouldn't before? I guess the only time this would be an issue is if you were for some reason retrieving job off of repo and then re-applying to a different Definitions object? Which I don't see ever happening.

Tl;dr I think it's fine